sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_link_async.py

```diff
@@ -273,5 +274,19 @@ async def detach(self, close: bool = False, error: Optional[AMQPError] = None) -
             await self._set_state(LinkState.DETACHED)
 
     async def flow(self, *, link_credit: Optional[int] = None, **kwargs) -> None:
-        self.current_link_credit = link_credit if link_credit is not None else self.link_credit
-        await self._outgoing_flow(**kwargs)
+        # Given the desired link credit `link_credit`, the link credit sent via
+        # FlowFrame is calculated as follows: The link credit to flow on the wire
+        # `self.current_link_credit` is the desired link credit `link_credit`
+        # minus the current link credit on the wire `self.total_link_credit`.
+        self.current_link_credit = link_credit - self.total_link_credit \
```

Comment:
is there a risk of self.current_link_credit ending up negative if link_credit is smaller than total_credit?

Reply:
yes I believe it does go negative. follow up: does this affect sender flow at all?

Reply:
yes there is a barrier on outgoing_flow() to prevent a negative flow from being sent

Reply:
everywhere else (except for in session _incoming_transfer) current_link_credit is set to something else before it calls _outgoing_flow